ci/build-all-defconfigs.sh

```diff
@@ -100,6 +105,16 @@ function build_sdk
 		--disable ROOTFS_DEVICE_CREATION_DYNAMIC_EUDEV \
 		--enable INSTALL_LIBSTDCPP
 
+	# Enable toolchains we'll need to be built as part of the SDK, and make sure we
+	# consider them to make the sdk unique
+	buildroot/utils/config --file $SDK_BUILD_DIR/.config --package \
+		--enable P8_PORE_TOOLCHAIN  --enable HOST_P8_PORE_BINUTILS \
+		--enable PPE42_TOOLCHAIN --enable HOST_PPE42_GCC --enable HOST_PPE42_BINUTILS
```

Comment:
I'm trying to work out why this is needed and not caught by the defconfig that the SDK is being built for.

Is this a remnant of my old hack where "assume SDK for platform X is okay for the rest of them" that was part of the Jenkins CI setup before I left IBM?

Reply:
> I'm trying to work out why this is needed and not caught by the defconfig that the SDK is being built for.

In a witherspoon_defconfig, ppe42-toolchain would already be selected. In a garrison_defconfig, p8-pore-toolchain. In a zz_defconfig, none of them would be selected.
The idea was to try forcibly select them to make the SDK as reusable as possible, so we have a single sdk package capable of building both witherspoon as well as garrisons (but not zz - see below)

> Is this a remnant of my old hack where "assume SDK for platform X is okay for the rest of them" that was part of the Jenkins CI setup before I left IBM?
It's actually quite the opposite.

If you use pristine _defconfig, we will build a SDK that will be the representation of the toolchain that was used by that particular _defconfig:

- witherspoon_defconfig builds a gcc 6.5.0 with C++ support (for hostboot)
- zz_defconfig builds a gcc 7.4.0 without C++ support (i.e., hostboot is not required)

This was causing hazards such as zz being built with gcc 7.4.0 when built back-to-back, and using gcc 6.5.0 when using the (witherspoon-based) sdk. This would be theoretically fine, but for CI consistency purposes, I decided to rework that with #3586 .